### PR TITLE
[PORT] Synths

### PR DESCRIPTION
**Pull request by RimiNosha**
## About The Pull Request

I want to fucking stab organ code with a rusty knife I swear to god.

Adds synths as they appear in Skyrat with a couple of changes:

Screeen code is no longer species dependent, instead anyone can implant a screen onto themselves.
There are no racial drinks. Yet to be added.

## How Does This Help ***Gameplay***?

One of our core races.

## How Does This Help ***Roleplay***?

Robit go beep boop.

## Credits
Azarak - Initial code.
Gandalf2k15 - Updating some code on SR.
Nerevar - Updating some code on SR.
RimiNosha - Updating the code and adding various QoL features, both here and SR.

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary> 

image
image
image
image
image
image
image
image

</details>

## Changelog

:cl:
add: Synths, check them out!
/:cl:

### CODE REVIEW

File: code/modules/synths/code/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+### Defines:
+
+- Will fill out as I discover what defines were made to acommodate these.
+
+### Included files that are not contained in this module:
+
+- N/A
+
+### Credits:
+Nerevar - Initial code, I think. Correct this file if wrong.
```
Comment (nevimer) [inline code/modules/synths/code/README.md:28]: Azarak, Gandalf.
https://github.com/Skyrat-SS13/Skyrat-tg/pull/847

Comment (nevimer) [inline code/modules/synths/code/README.md:28]: Was made by Azarak, polished by Gandalf initially. 
https://github.com/Skyrat-SS13/Skyrat-tg/pull/675
https://github.com/Skyrat-SS13/Skyrat-tg/pull/847
https://github.com/Skyrat-SS13/Skyrat-tg/pull/1000

**Review by nevimer — COMMENTED** [review]
So, I tested this. Currently, synths seem to lack snouts, horns, tails (other than cat) ears. I also am stuck with the colour pink, in the current state for chassis that don't have a pre-sprited colour?

Functionality-wise, it all works in the game. No runtimes, mechanics work as expected. I tested. 

The species features menu looks a little weird. 

the modules/synths/code/surgery folder could be called synths instead of robots - for consistency. Shouldn't the surgery folder be in the surgery module? For ease of navigation, as my first place to look for all the games surguries would be in the surgery module not the synth module.  

Finally, the attribution for the OG synth code belongs to Azarak for the commented PR's on Skyrat. 

This is review pass one.

**Comment by nevimer** [comment]
I36ZLpRZXV

File: code/modules/client/preferences/species_features/synth.dm [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+	if(!issynthetic(target))
+		return
+
+	var/obj/item/organ/internal/brain/new_brain
+
+	switch(value)
+		if(ORGAN_PREF_CIRCUIT_BRAIN)
+			new_brain = /obj/item/organ/internal/brain/synth/circuit
+		if(ORGAN_PREF_MMI_BRAIN)
+			new_brain = /obj/item/organ/internal/brain/synth/mmi
```
[the remark this answers is not in the document]
Comment (RimiNosha) [reply code/modules/client/preferences/species_features/synth.dm:27]: Not something I'm going to be dealing with in this PR, if at all. The amount of snowflake code required for that would be unreasonable in my eyes.

File: code/modules/mob/living/carbon/human/dummy.dm [hunk 3]
```diff
@@ -95,6 +94,7 @@ INITIALIZE_IMMEDIATE(/mob/living/carbon/human/dummy)
 	target.dna.features["tail"] = "None"
 	target.dna.features["tail_lizard"] = "Smooth"
 	target.dna.features["pod_hair"] = "Ivy"
+	target.dna.features[MUTANT_SYNTH_SCREEN] = "None"
```
[the remark this answers is not in the document]
Comment (RimiNosha) [reply code/modules/mob/living/carbon/human/dummy.dm:97]: Too bad. TG for some godforsaken reason refuses to use defines, but I'd rather just avoid the possibility of typos from the get go.

File: code/modules/synths/code/species/synthetic.dm [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+	brutemod = 1.3
+	coldmod = 1.2
+	heatmod = 2 // TWO TIMES DAMAGE FROM BEING TOO HOT?! WHAT?! No wonder lava is literal instant death for us.
+	siemens_coeff = 1.4 // Not more because some shocks will outright crit you, which is very unfun
+
+/datum/species/synthetic/spec_life(mob/living/carbon/human/human)
+	if(human.stat == SOFT_CRIT || human.stat == HARD_CRIT)
+		human.adjustFireLoss(1) //Still deal some damage in case a cold environment would be preventing us from the sweet release to robot heaven
+		human.adjust_bodytemperature(13) //We're overheating!!
+		if(prob(10))
```
[the remark this answers is not in the document]
Comment (RimiNosha) [reply code/modules/synths/code/species/synthetic.dm:69]: It's a prob. It'll be wildly inconsistent in how fast it murders.

File: code/modules/synths/code/species/prefsmenu.dm [hunk 5]
```diff
@@ -0,0 +1,5 @@
+/datum/species/synthetic/get_species_description()
+	return null
```
[the remark this answers is not in the document]
Comment (RimiNosha) [reply code/modules/synths/code/species/prefsmenu.dm:2]: TGUi doesn't care.

File: code/modules/synths/code/species/synthetic.dm [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+			to_chat(human, span_warning("Alert: Critical damage taken! Cooling systems failing!"))
+			do_sparks(3, TRUE, human)
+
+/datum/species/synthetic/on_species_gain(mob/living/carbon/human/transformer)
+	. = ..()
+	var/obj/item/organ/internal/appendix/appendix = transformer.getorganslot(ORGAN_SLOT_APPENDIX)
+	if(appendix)
```
[the remark this answers is not in the document]
Comment (RimiNosha) [reply code/modules/synths/code/species/synthetic.dm:76]: No.

**Comment by RimiNosha** [comment]
Synths have their own colour selectors for everything. I need to add the flag for no skin colors in code.

Snouts and horns don't show because I haven't added them yet, I've a couple of other things to fix first before I risk breaking even more.

Basic species descriptions will be made before this PR will be merged, and I'll be doing a pass of the species features that are listed.

**Comment by RimiNosha** [comment]
Alright, done what I had on my list.

**Comment by nevimer** [comment]
I want to merge this, but CI must be happy

**Pull request merged**